clients/drcachesim/common/options.cpp

```diff
@@ -333,3 +333,15 @@ droption_t<bool> op_reuse_verify_skip
  "Verifies every skip list-calculated reuse distance with a full list walk. "
  "This incurs significant additional overhead.  This option is only available "
  "in debug builds.");
+
+// XXX: replace function return address with function callstack
+droption_t<std::string> op_record_function
+(DROPTION_SCOPE_FRONTEND, "record_function", DROPTION_FLAG_ACCUMULATE, ";",
+ std::string("malloc|0|1") + ";" + std::string("tc_malloc|1|1") + ";" +
+ std::string("__libc_malloc|2|1"),
+ "Record invocation traces for the specified functions",
+ "Option value should fit this format"
+ "function_name|function_id|function_args_num (example:"
+ " -record_function \"tc_malloc|2|1\"). The trace would contain information for"
+ " function return address, function argument value(s), and function return"
+ " value. We only record pointer-sized arguments and return value.");
```

Comment:
I think we should add a section to drcachesim.dox.in on function tracing.  Fine to be in a separate CL if you like to avoid dragging this one out further.

Reply:
Will make a separate CL for this, and I will put a XXX coment about this.

Reply:
Done